noir-projects/noir-protocol-circuits/crates/private-kernel-lib/src/components/previous_kernel_validator.nr

```diff
@@ -61,12 +78,4 @@ impl PreviousKernelValidator {
             array_length(self.previous_kernel.validation_requests.scoped_key_validation_requests_and_generators), 0, "Non empty key validation requests"
         );
     }
-
-    fn verify_no_transient_data(self) {
```

Comment:
Removing this check as it's expensive to check if the remaining nullifiers that link to note hashes are revertible and those note hashes are non-revertible.
Which means it's now ok to not squash transient note hashes and nullifiers that can be squashed. The user will have to pay more fees but it should be fine in terms of security and privacy.